src/lib/qnio/utils.c

```diff
@@ -9,6 +9,117 @@
  */
 
 #include "defs.h"
+#include "qnio.h"
+
+int 
+is_secure()
+{
+    if (access(SECURE_IMPL, F_OK) != 0)
```

Comment:
This is all general best practice for library design. Library code in general should not rely on configuration files to determine its behaviour as different concurrent users of the library may wish to have different configuration settings. Use of external configuration files should be delegated to users of the library, or if the library does wish to have some defaults loaded from config files, it should always allow those defaults to be overridden via API calls.